File: Java/1 - Two Sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer,Integer> hm = new HashMap<Integer,Integer>();
        for(int i = 0; i < nums.length; i++) {
            int required = target - nums[i]; 
            if(hm.containsKey(required)) {
                int[] arr = {hm.get(required), i};
                return arr; 
            }
            hm.put(nums[i], i);
        }
        return null;
    }
}

File: Python/1 - Two Sum.py
class Solution:
    def twoSum(self, nums: list[int], target: int) -> list[int]:
        index = {}  # Dictionary to store the indices of numbers

        for i, num in enumerate(nums):
            #enumerate returns index and value
            complement = target - num
            if complement in index:
                return [index[complement], i]
            index[num] = i
